### PR TITLE
Created game logic

### PR DESCRIPTION
**Pull request by BenjaminGarza**

### CODE REVIEW

File: -lib-/logic.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
+class Player
+    attr_accessor :character, :name
```
Comment (AdnanAfsari) [inline -lib-/logic.rb]: Use attr_readers/attr_writers instead of atrr_acessors.
make this change where necessary

File: -lib-/logic.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        @character = character
+        @name = name
+    end
+
+   end
+class Gameboard
+attr_accessor  :current_state, :current_player, :winner
+
+   def initialize
+    @current_state = [["Empty","Empty","Empty"],["Empty","Empty","Empty"],["Empty","Empty","Empty"]]
```
Comment (AdnanAfsari) [inline -lib-/logic.rb]: make it looks like a real board, Empty does not make any sense i guess

File: -lib-/logic.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    @current_player = true
+    @winner = false
+    end
+
+    def win_check(player_character)
```
Comment (AdnanAfsari) [inline -lib-/logic.rb]: Its a long method, is there any way we can break it down???

File: -lib-/logic.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    def turn(current_move, player_character)
+    @current_move = current_move.split(",")
+    @x = @current_move[0].to_i - 1
+    @y = @current_move[1].to_i - 1
+
+    if @current_state[@x][@y] == "Empty"
+        @current_state[@x][@y] = player_character
+    else
+        return puts "Please make a valid move"
```
Comment (AdnanAfsari) [inline -lib-/logic.rb]: game logic should not contain any user interactions like puts , prints , main.rb should be responsible it.

**Review by AdnanAfsari — CHANGES_REQUESTED** [review]
Hello @BenjaminGarza @mapra99 ✋ 

Great effort guys nice work but you have to make some improvement and changes to get approval.Please check my comments.

- [x] First of all, I tried to play your game but it breaks.have a look in the screenshot.
Screenshot 2019-08-28 at 3 46 27 AM

- [x] Your board does not looks good, You better show numbers for every positions instead of "empty"

- [x] When i select position 1, it select top right most corner in the board which meant to be number 3.
make it select the right positions.

- [x] The project uses at least two classes (Usually it uses three: Game, Board and Player. The code of the project should be logically divided between classes.

- [x] Your main.rb does not contain any method, Please divide  responsibilities into methods.

Please make those changes for now and make your game work then  submit for another review 😄 

Happy Coding

File: lib/logic.rb [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    def initialize(character,name)
+        @character = character
+        @name = name
+    end
+
+   end
+
+class Board
+  attr_accessor :current_state
```
Comment (maelfosso) [inline lib/logic.rb]: Use `attr_reader`instead.

File: bin/main.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+load '../lib/logic.rb'
```
Comment (maelfosso) [inline bin/main.rb]: Use `require_relative` please
Here what I have when run it with `load`
```ruby
$ ruby bin/main.rb 
Traceback (most recent call last):
	1: from bin/main.rb:1:in `<main>'
bin/main.rb:1:in `load': cannot load such file -- ../lib/logic.rb (LoadError)
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

Really, really original game. It's very very different from the ones I played .
Almost done. Check the comment below and kindly make the corresponding changes, please.

Then re-submit a code review request.

**Comment by Oghenebrume50** [comment]
reviewing......

File: lib/logic.rb [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    end
+
+    def show
+    (0..2).each do |i|
+    (0..2).each do |j|
+        print @board.current_state[i][j]
+        print "\t"
+    end
+    print "\n"
+    end
```
Comment (Oghenebrume50) [inline lib/logic.rb]: all display should be handled in the main.rb

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
Hi team, so this is working, but could you fix the outlined issue and file for a re-review? :+1:

**Review by Ispirett — APPROVED** [review]
Great!! works your approach is different I like that :+1: :+1: :1st_place_medal: 

#### Optional not required to complete this project
- [ ] You should put the game instructions at the start of the game 

#### Optional not required to complete this project
- [ ]  Modualizing your code Is a good practice, you can make a separate file for each class and use `require_relative` to import them

#### Also please fix conflicts before merge

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello coders,

The game has evolved. Really Really !!!
Before moving forward, please,  let me show you some bugs to fix.

Take a look in that execution, please
```ruby
$ ruby bin/main.rb 
Enter player X name
Yu
Enter player O name
Pi
1,1	2,1	3,1	
1,2	2,2	3,2	
1,3	2,3	3,3	
Yu VS Pi
Make a move Yu
1,1
X	2,1	3,1	
1,2	2,2	3,2	
1,3	2,3	3,3	
Make a move Pi
2,2
X	2,1	3,1	
1,2	O	3,2	
1,3	2,3	3,3	
Make a move Yu
2,1
X	X	3,1	
1,2	O	3,2	
1,3	2,3	3,3	
Make a move Pi
3,1
X	X	O	
1,2	O	3,2	
1,3	2,3	3,3	
Make a move Yu
1,3
X	X	O	
1,2	O	3,2	
X	2,3	3,3	
Make a move Pi
1,2
X	X	O	
O	O	3,2	
X	2,3	3,3	
Make a move Yu
3,3
X	X	O	
O	O	3,2	
X	2,3	X	
Make a move Pi
2,3
X	X	O	
O	O	3,2	
X	O	X	
Make a move Yu
3,2
X	X	O	
O	O	X	
X	O	X	
Pi is the victor!
```
I was expecting a `Draw` but I instead go *Pi is the victor!*
Did I do something wrong, please ?

When I changed that section of code, it solved the problem
https://github.com/BenjaminGarza/Tic-tac-toe/blob/422510a707b24839aad9d08a2b323eab3ce68559/bin/main.rb#L39-L45

Kindly check it too and make the corresponding changes then submit a new code review request

**Comment by Oghenebrume50** [comment]
Reviewing.......

**Review by Oghenebrume50 — APPROVED** [review]
hi team kudos on the changes so far,  the game now works perfectly. You can move on know :smiley:

**Pull request merged**